.storybook/main.ts

```diff
@@ -0,0 +1,25 @@
+import type { StorybookConfig } from '@storybook/react-vite';
+import { mergeConfig } from 'vite';
+
+import viteConfig from '../vite.config';
+
+const config: StorybookConfig = {
+  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
+  addons: [
+    '@storybook/addon-links',
+    '@storybook/addon-essentials',
+    '@storybook/addon-interactions',
+  ],
+  framework: {
+    name: '@storybook/react-vite',
+    options: {},
+  },
+  docs: {
+    autodocs: 'tag',
+  },
+  async viteFinal(config) {
+    // Merge custom configuration into the default config
+    return mergeConfig(config, viteConfig);
+  },
```

Comment:
It's less readable IMHO; it will need changing to what it already is later if someone needs to modify it. I am not a big fan of fewer lines just because, and definitely not of arrow functions without parens.

Reply:
async is not currently necessary, but any person contributing to that method will know, without having to check storybook docs, that it supports async operations. I'm gonna leave it